ToolBox/AlgoSeekOptionsConverter/InMemoryDataProcessor.cs

```diff
@@ -0,0 +1,141 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using QuantConnect.Data;
+using QuantConnect.Util;
+using System.Collections.Generic;
+using QuantConnect.Data.Consolidators;
+using System.IO;
+
+namespace QuantConnect.ToolBox.AlgoSeekOptionsConverter
+{
+	/*
+	 * Possible Replacement:
+	 * https://gist.github.com/AnObfuscator/a619eea8e9b66ef4967bd2c679400d39
+	 */
+	public class InMemoryDataProcessor : IDataProcessor
+	{
+		private readonly TickType _tickType;
+		private readonly ConcurrentDictionary<Symbol, TickContainer> _ticksBySymbol;
```

Comment:
Please make this not a concurrent dictionary - it could be hiding design flaws. There should be no way with a single threaded streaming design one tick after another that it can simultaneously add to the dictionary from two places.